logstash-core/src/test/java/org/logstash/ackedqueue/HeadPageTest.java

```diff
@@ -1,104 +1,98 @@
 package org.logstash.ackedqueue;
 
 import org.junit.Test;
-import org.logstash.ackedqueue.io.ByteBufferPageIO;
 import org.logstash.ackedqueue.io.PageIO;
 
 import java.io.IOException;
 
 import static org.hamcrest.CoreMatchers.equalTo;
 import static org.hamcrest.CoreMatchers.is;
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.logstash.ackedqueue.QueueTestHelpers.singleElementCapacityForByteBufferPageIO;
 
 public class HeadPageTest {
 
     @Test
     public void newHeadPage() throws IOException {
         Settings s = TestSettings.volatileQueueSettings(100);
-        Queue q = new Queue(s);
-        PageIO pageIO = s.getPageIOFactory().build(0, 100, "dummy");
-        pageIO.create();
-        HeadPage p = new HeadPage(0, q, pageIO);
-
-        assertThat(p.getPageNum(), is(equalTo(0)));
-        assertThat(p.isFullyRead(), is(true));
-        assertThat(p.isFullyAcked(), is(false));
-        assertThat(p.hasSpace(10), is(true));
-        assertThat(p.hasSpace(100), is(false));
-
-        q.close();
+        try(Queue q = new Queue(s)) {
+            PageIO pageIO = s.getPageIOFactory().build(0, 100, "dummy");
+            pageIO.create();
+            HeadPage p = new HeadPage(0, q, pageIO);
+
+            assertThat(p.getPageNum(), is(equalTo(0)));
+            assertThat(p.isFullyRead(), is(true));
+            assertThat(p.isFullyAcked(), is(false));
+            assertThat(p.hasSpace(10), is(true));
+            assertThat(p.hasSpace(100), is(false));
+        }
     }
 
     @Test
     public void pageWrite() throws IOException {
         Queueable element = new StringElement("foobarbaz");
-        int singleElementCapacity = ByteBufferPageIO.HEADER_SIZE + ByteBufferPageIO._persistedByteCount(element.serialize().length);
 
-        Settings s = TestSettings.volatileQueueSettings(singleElementCapacity);
-        Queue q = new Queue(s);
-        q.open();
-        HeadPage p = q.headPage;
-
-        assertThat(p.hasSpace(element.serialize().length), is(true));
-        p.write(element.serialize(), 0, 1);
+        Settings s = TestSettings.volatileQueueSettings(singleElementCapacityForByteBufferPageIO(element));
+        try(Queue q = new Queue(s)) {
+            q.open();
+            HeadPage p = q.headPage;
```

Comment:
I don't _think_ so in this case - we are not creating the headPage, but using the one inside the queue, which is closed within queue.close() which is called via the try... with resources